bridge-impl/src/main/java/com/liferay/faces/bridge/context/url/internal/BridgeURLBase.java

```diff
@@ -0,0 +1,714 @@
+
+		// For each parameter found in the original query-string:
+		for (String queryParameter : queryParameters) {
+
+			if ((queryParameter != null) && (queryParameter.length() > 0)) {
+
+				// Parse the name and value from the name=value pair.
+				String[] nameValueArray = queryParameter.split("[=]");
+
+				String name = null;
+				String[] values = null;
+
+				if (nameValueArray.length == 1) {
+
+					name = nameValueArray[0].trim();
+					values = new String[] { "" };
+				}
+				else if (nameValueArray.length == 2) {
+
+					name = nameValueArray[0].trim();
+
+					// If the parameter name is present in the parameter map, then that means it should be appended
+					// to the return value. Otherwise, it should not be appended, because absence from the parameter
+					// map means that it was deliberately removed.
+					values = parameterMap.get(name);
+				}
+
+				if ((name == null) || (name.length() == 0)) {
+					logger.error("Invalid name=value pair=[{0}] in URL=[{1}]: name cannot be empty", queryParameter,
+						url);
+				}
+				else if ((values == null) || (values.length == 0)) {
+
+					// Note that "javax.portlet.faces.BackLink" is sometimes deliberately removed and therefore is
+					// not an error.
+					if (!Bridge.BACK_LINK.equals(name)) {
+						logger.error("Invalid name=value pair=[{0}] in URL=[{1}]", queryParameter, url);
+					}
+				}
+				else {
+
+					if (firstParam) {
+						firstParam = false;
+					}
+					else {
+						buf.append("&");
+					}
+
+					buf.append(name);
+					buf.append("=");
+
+					Integer parameterOccurrences = parameterOccurrenceMap.get(name);
+
+					if (parameterOccurrences == null) {
+						parameterOccurrences = new Integer(0);
+					}
+
+					String value = values[parameterOccurrences.intValue()];
+					buf.append(value);
+					parameterOccurrences = new Integer(parameterOccurrences.intValue() + 1);
+					parameterOccurrenceMap.put(name, parameterOccurrences);
+				}
+			}
+		}
+
+		// Now iterate through the entire parameter map to see and append any additional parameters to the return
+		// value.
+		Set<Map.Entry<String, String[]>> mapEntries = parameterMap.entrySet();
+
+		for (Map.Entry<String, String[]> mapEntry : mapEntries) {
+
+			String name = mapEntry.getKey();
+
+			if (parameterOccurrenceMap.get(name) == null) {
+				String[] values = mapEntry.getValue();
+
+				if ((values != null) && (values.length > 0)) {
+
+					if (firstParam) {
+						firstParam = false;
+					}
+					else {
+						buf.append("&");
+					}
+
+					buf.append(name);
+					buf.append("=");
+					buf.append(values[0]);
+				}
+			}
+		}
+
+		if (fragmentId != null) {
+			buf.append(fragmentId);
+		}
+
+		return buf.toString();
+	}
+
+	@Override
+	public final String removeParameter(String name) {
+
+		Map<String, String[]> parameterMap = getParameterMap();
+		String[] values = parameterMap.remove(name);
+
+		if (values == null) {
+			values = parameterMap.remove(namespace + name);
```

Comment:
@ngriffin7a, I've changed the implementation of this method.

[Old](https://github.com/liferay/liferay-faces-bridge-api/pull/7/files#diff-aa0af658d0ff29f5ceade16735f00d61L485):

```
protected String removeParameter(String name) {

	String[] values = getParameterMap().remove(name);
	String value = null;

	if ((values != null) && (values.length > 0)) {
		value = values[0];
	}

	return value;
}
```

New:

```
@Override
public final String removeParameter(String name) {

	Map<String, String[]> parameterMap = getParameterMap();
	String[] values = parameterMap.remove(name);

	if (values == null) {
		values = parameterMap.remove(namespace + name);
	}

	String value = null;

	if ((values != null) && (values.length > 0)) {
		value = values[0];
	}

	return value;
}
```

I added a check for a namespaced version of the parameter.